scripts/environment-unpinned-dev.yaml

```diff
@@ -5,14 +5,16 @@ dependencies:
     - flake8
     - flake8-isort
     - ipython
-    - localstack
+    - localstack==0.11.1.1
     - pip
     - pre-commit
     - pytest
     - pip:
         # Select depenedencies from localstack[full] that we need
-        - amazon-kclpy
-        - crontab
-        - flask-cors
-        - moto-ext
-        - subprocess32
+        # These should be pinned to compatible versions for the current
+        # localstack version
+        - amazon-kclpy==2.0.1
+        - crontab==0.22.6
+        - flask-cors==3.0.8
+        - moto-ext==1.3.15.15
+        - subprocess32==3.5.4
```

Comment:
i don't believe it's necessary to pin all of these dependencies. but this is a pure gut instinct reaction.

Reply:
My concern is that overpinning creates future tech debt when packages should be updated to more secure / uptodate versions but we don't know / remember why packages were pinned.

Reply:
But your comment is clear, so I'm "meh" about it in this case.

Reply:
I take above comments back. I no longer think its necessary to pin localstack at all.

Reply:
I narrowed it down to `moto-ext` (at least when I run the tests locally)

I was also wondering whether to pin all of them or just that one. I chose to pin all of them because it took me a while for me to recognize that this was the issue. localstack didn't give a helpful error, and I didn't initially realize that we added some of localstack's dependencies separate from localstack itself. Let's say all of these packages were updated in the future, the user would have to test them 1 by 1 to find the responsible package (or set of packages).

The risk I see if the pinned versions are incompatible with other (non-pinned) python packages. And of course security vulnerabilities, but hopefully GitHub will warn us there and we can update as needed.

I'm secretly hoping we don't have to keep this around too long and can just get rid of localstack (https://github.com/mozilla/OpenWPM/issues/652).

wdyt?

Reply:
I don't see a reason to pin localstack, so then pinning moto-ext doesn't seem necessary.

Reply:
@birdsarah Why not? Pinning these dependencies fixed the test timeouts we were seeing here: https://travis-ci.org/github/mozilla/OpenWPM/jobs/699484251#L1132-L1158.

It's not clear to me _why_ pinning fixed the timeout. I had assumed it was a dependency mismatch, but the newer version of `moto-ext` is valid according to localstack's [requirement.txt](https://github.com/localstack/localstack/blob/6dcffb35fb685b4297599c9ee1064b6433ab64c9/requirements.txt#L36).

Reply:
Pinning didn't fix the timeout. The timeout was a fluke. I re-ran the tests that failed before you pinned and they passed.

Reply:
Oh it looks like I missed your comment here: https://github.com/mozilla/OpenWPM/pull/682#issuecomment-646220583. It seems like this build passed https://travis-ci.org/github/mozilla/OpenWPM/builds/699484265. Interesting! I could reproduce the timeouts locally as well. Let me investigate.

Reply:
Hmmmmm....They're timing out here too: https://travis-ci.org/github/mozilla/OpenWPM/jobs/699484251

Reply:
There's two runs associated with each commit - I only successfully re-ran the tests on one. I don't know whether there's a meaningful difference between the two runs.

Reply:
OK. My apologies. I now see the difference between the two runs. Please ignore all my comments regarding pinning. Glad you've pinned it down to moto-ext. Does this mean we can advance localstack and just pin moto-ext?

Reply:
> Glad you've pinned it down to moto-ext. Does this mean we can advance localstack and just pin moto-ext?

Yes looks like it. I just tried it locally and that worked. I'll push a new commit.

Reply:
okay I spoke too soon. I was bitten by my local packages. There are still failures on travis https://travis-ci.org/github/mozilla/OpenWPM/jobs/701020698. I've now re-pinned localstack.